src/DGtal/topology/CubicalComplex.h

```diff
@@ -0,0 +1,1305 @@
+/**
+*  This program is free software: you can redistribute it and/or modify
+*  it under the terms of the GNU Lesser General Public License as
+*  published by the Free Software Foundation, either version 3 of the
+*  License, or  (at your option) any later version.
+*
+*  This program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+**/
+
+#pragma once
+
+/**
+* @file CubicalComplex.h
+* @author Jacques-Olivier Lachaud (\c jacques-olivier.lachaud@univ-savoie.fr )
+* Laboratory of Mathematics (CNRS, UMR 5127), University of Savoie, France
+*
+* @date 2015/08/28
+*
+* Header file for module CubicalComplex.cpp
+*
+* This file is part of the DGtal library.
+*/
+
+#if defined(CubicalComplex_RECURSES)
+#error Recursive header files inclusion detected in CubicalComplex.h
+#else // defined(CubicalComplex_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define CubicalComplex_RECURSES
+
+#if !defined CubicalComplex_h
+/** Prevents repeated inclusion of headers. */
+#define CubicalComplex_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <boost/type_traits.hpp>
+#include <boost/iterator/iterator_facade.hpp>
+#include "DGtal/base/Common.h"
+#include "DGtal/base/ConstAlias.h"
+#include "DGtal/base/Alias.h"
+#include "DGtal/base/ContainerTraits.h"
+#include "DGtal/topology/CCellularGridSpaceND.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  /**
+  * Any cell is stored within a cubical complex with an associated
+  * data, which must derive from this class. Its basic usage is to
+  * store flags associated to the cells, but it may store other
+  * values.
+  *
+  * @note Predefined flags are CubicalComplex::REMOVED,
+  * CubicalComplex::COLLAPSIBLE, CubicalComplex::FIXED,
+  * CubicalComplex::USER1. Other bits can be used to associate an
+  * integer to the cell. The corresponding mask is
+  * CubicalComplex::VALUE. 
+  *
+  * @note Such data is notably used in collapse operation
+  * (CubicalComplex::collapse).
+  */
+  struct CubicalCellData {
+    inline CubicalCellData() : data( 0 ) {}
+    CubicalCellData( uint32_t d ) : data( d ) {}
+    uint32_t data;
+  };
+  
+  template < typename TKSpace, typename TCellContainer >
+  class CubicalComplex;
+  
+  namespace functions {
+    namespace ccops {
+      template < typename TKSpace, typename TCellContainer >
+      CubicalComplex< TKSpace, TCellContainer >&
+      operator|=( CubicalComplex< TKSpace, TCellContainer >&,
```

Comment:
Since these operators are related to ANR explicit (but templated) class Cubicalcomplex, i don't see the interest o move the operators to a specific namespace. (The set functions cases is a bit different since we may also use == operator on std vectors without considering them as set containers)